src/storage/datacore/restapi/mode/poolspaceusage.pm

```diff
@@ -0,0 +1,121 @@
+use JSON::XS;
+
+sub new {
+    my ($class, %options) = @_;
+
+    my $self = $class->SUPER::new(package => __PACKAGE__, %options, force_new_perfdata => 1);
+    bless $self, $class;
+    $options{options}->add_options(arguments => {
+        'pool-id:s' => { name => 'pool_id' } });
+
+    return $self;
+}
+
+sub check_options {
+    my ($self, %options) = @_;
+    $self->SUPER::check_options(%options);
+    if (centreon::plugins::misc::is_empty($self->{option_results}->{pool_id})) {
+        $self->{output}->add_option_msg(short_msg => 'Please set pool-id option');
+        $self->{output}->option_exit();
+    }
+}
+
+sub set_counters {
+    my ($self, %options) = @_;
+
+    $self->{maps_counters_type} = [
+        # BytesAllocatedPercentage is the disk usage of the pool
+        # in datacore you can make thin provisionning, where you give to each partition more than what you really have.
+        # oversubscribed is the number of Bytes allocated minus the number of bytes present in the system.
+        { name => 'BytesAllocatedPercentage', type => 0 },
+        { name => 'oversubscribed', type => 0 },
+    ];
+    $self->{maps_counters}->{BytesAllocatedPercentage} = [
+
+        {
+            label  => 'bytesallocatedpercentage',
+            nlabel => 'datacore.pool.bytesallocated.percentage',
+            set    => {
+                key_values      => [ { name => 'bytesallocatedpercentage' } ],
+                output_template => 'Bytes Allocated : %s %%',
+                perfdatas       => [
+                    { template => '%d', unit => '%', min => 0, max => 100 }
+                ]
+            }
+        } ];
+    $self->{maps_counters}->{oversubscribed} = [ {
+        label  => 'oversubscribed',
+        nlabel => 'datacore.pool.oversubscribed.bytes',
+        set    => {
+            key_values      => [ { name => 'oversubscribed' } ],
+            output_template => 'Over subscribed bytes : %s',
+            perfdatas       => [
+                { template => '%d', unit => 'bytes', min => 0 }
+            ]
+        }
+    } ];
+}
+
+sub manage_selection {
+    my ($self, %options) = @_;
+
+    my $data = $options{custom}->request_api(
+        url_path => '/RestService/rest.svc/1.0/performances/' . $self->{option_results}->{pool_id},
+    );
+    if (defined($data->[1])) {
+        $self->{output}->add_option_msg(short_msg => 'multiples pools found in api response, only one is expected. Please check pool_id and datacore versions.');
+        $self->{output}->option_exit();
+    }
+    $self->{BytesAllocatedPercentage}->{bytesallocatedpercentage} = $data->[0]->{"BytesAllocatedPercentage"};
+    $self->{oversubscribed}->{oversubscribed} = $data->[0]->{"BytesOverSubscribed"};
+
+}
+1;
+
+__END__
+
+=head1 MODE
+
+Check Datacore pool space and over subscribed usage exposed through the rest api.
+
+=over 8
+
+=item B<--pool-id>
+
+Id of the pool to check. See list-pool autodiscovery mode to list pools id (required).
```

Comment:
I'm surprise about this usage of a mandatory option. I'm not sure to already have seen a mode using filter after a list.pm usage mandatory.
I think It could be better to monitor all pools and use a filter-name if user want to monitor a specific one.
Something like [this](https://github.com/centreon/centreon-plugins/blob/c245c4104160636d900256ba0abd1bc836437467/src/apps/ceph/restapi/mode/pools.pm#L59) with a global counter of pools and for each pools all the informations collected.
This should change a lot this mode configuration.

Reply:
As discuted together, this mode make a request in the form host/api/${id} so we can't get every value in one request.